Source/1.3/AI/AIFacilityManager.cs

```diff
@@ -50,111 +43,110 @@ public HashSet<FacilityDef> FacilityDefsInstalled
                     {
-                        cachedFacilitiesDef.AddRange(manager.FacilityDefsInsalled);
+                        cachedFacilityDefs.AddRange(manager.FacilityDefsInstalled);
                     }
                 }
-                return cachedFacilitiesDef;
+
+                return cachedFacilityDefs;
             }
         }
 
         /// <summary>
-        /// Find a low resource and attempt to build a facility for it.
+        ///     Find a low resource and attempt to build a facility for it.
         /// </summary>
-        /// <returns></returns>
+        /// <returns>Whether a <see cref="Facility" /> was successfully built</returns>
         public bool BuildResourceFacility()
         {
             List<ResourceDef> resourceDefs = player.ResourceManager.FindLowResources();
             ResourceDef def = resourceDefs.RandomElement();
 
-            if (FacilityDefsInstalled.Any(x=>x.ProducedResources.Contains(def)))
-            {
-                FacilityDef facilityDef = FacilityDefsInstalled.Where(x=>x.ProducedResources.Contains(def)).RandomElement();
-                return BuildNewFacility(facilityDef);
-            }
-            return false;
+            FacilityDef facilityDef = FacilityDefsInstalled.Where(
+                facility => facility.ProducedResources.Contains(def)).RandomElementWithFallback();
+
+            return facilityDef != null && BuildNewFacility(facilityDef);
         }
 
 
         /// <summary>
-        /// Attempt to build a facility
+        ///     Attempt to build a facility
         /// </summary>
-        /// <param name="facility"></param>
+        /// <param name="facilityDef"></param>
         /// <returns></returns>
         public bool BuildNewFacility(FacilityDef facilityDef)
         {
-            bool hasRemovedAll = true;
-            KeyValuePair<Settlement, FacilityManager> settlementAndManager = player.Manager.Settlements.First(x => x.Value.CanBuildAt(facilityDef));
+            var hasRemovedAll = true;
+            KeyValuePair<Settlement, FacilityManager> settlementAndManager =
+                player.Manager.Settlements.First(x => x.Value.CanBuildAt(facilityDef));
 
             FacilityManager manager = settlementAndManager.Value;
             Settlement settlement = settlementAndManager.Key;
 
             if (settlement != null && manager != null)
             {
-
                 StorageTracker storageTracker = player.Manager.StorageTracker;
                 foreach (ThingDefCountClass thingDefCount in facilityDef.costList)
                 {
-                    hasRemovedAll = hasRemovedAll && storageTracker.CanRemoveThingsFromStorage(thingDefCount.thingDef, thingDefCount.count);
+                    hasRemovedAll = hasRemovedAll &&
+                                    storageTracker.CanRemoveThingsFromStorage(
+                                        thingDefCount.thingDef, thingDefCount.count);
```

Comment:
I think these line breaks are really excessive

Reply:
Changed line length to 200